### PR TITLE
feat(call-home): add helm chart components

### PR DESCRIPTION
**Pull request by niladrih**
Signed-off-by: Niladri Halder <niladri.halder26@gmail.com>

This PR adds helm chart template, values entries and README.md updates for call-home.

**Note:**
This PR can be merged only after the container image is push into the docker repository.

### CODE REVIEW

File: chart/templates/mayastor/call-home/call-home-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: call-home
+  template:
+    metadata:
+      labels:
+        app: call-home
+        openebs.io/logging: "true"
```
Comment (vharsh) [inline chart/templates/mayastor/call-home/call-home-deployment.yaml:17]: Do we want supportability tooling to capture this log by default?

File: chart/values.yaml [hunk 2]
```diff
@@ -414,3 +414,20 @@ loki-stack:
               - __meta_kubernetes_pod_uid
               - __meta_kubernetes_pod_container_name
               target_label: __path__
+
+call-home:
```
Comment (vharsh) [inline chart/values.yaml:418]: I need to check if hyphens are considered okay now in template key-strings, iirc this was not considered a good practice and helm has a function to allow templating the values where the keys with hyphens exist.

**Review by vharsh — COMMENTED** [review]
more changes might be required to making naming in sync

File: chart/templates/mayastor/call-home/call-home-deployment.yaml (hunk 1 above)
Comment (vharsh) [reply chart/templates/mayastor/call-home/call-home-deployment.yaml:17]: more than this, by default the Logging infra will capture & store it's logs by default for the set retention period

Comment (niladrih) [reply chart/templates/mayastor/call-home/call-home-deployment.yaml]: No, this component does not need logging. Good catch, thanks.

File: chart/values.yaml (hunk 2 above)
Comment (niladrih) [reply chart/values.yaml]: RIght! Good catch.

Comment (niladrih) [reply chart/values.yaml]: hyphens are not usable.

**Comment by niladrih** [comment]
Changed component name to obs-callhome from call-home

File: chart/templates/mayastor/obs/obs-callhome-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        - name: obs-callhome
+          image: "{{ .Values.image.registry }}/{{ .Values.image.repo }}/mayastor-obs-callhome:{{ .Values.image.tag }}"
+          args:
+            - "-e http://{{ .Release.Name }}-api-rest:8081"
+            - "-n {{ .Release.Namespace }}"
+          env:
+            - name: RUST_LOG
+              value: {{ .Values.obs.callhome.logLevel }}
+            {{- if default .Values.base.logSilenceLevel .Values.obs.callhome.logSilenceLevel }}
+            - name: RUST_LOG_SILENCE
```
Comment (tiagolobocastro) [inline chart/templates/mayastor/obs/obs-callhome-deployment.yaml:34]: Unless I'm mistaken, I don't think call home processes this env variable

**Review by tiagolobocastro — APPROVED** [review]

File: chart/templates/mayastor/obs/obs-callhome-deployment.yaml (hunk 3 above)
Comment (niladrih) [reply chart/templates/mayastor/obs/obs-callhome-deployment.yaml]: Right!

**Review by avishnu — APPROVED** [review]
lgtm

**Comment by niladrih** [comment]
bors try

**Comment by niladrih** [comment]
bors try

**Comment by niladrih** [comment]
bors try

**Comment by niladrih** [comment]
bors merge

**Pull request merged**